src/useStore.ts

```diff
@@ -209,6 +209,7 @@ export async function executeCode({
       errors = [e]
       logs = []
       if (e.msg === 'file is empty') engineCommandManager.endSession()
+      if (e instanceof KCLLexicalError) engineCommandManager.endSession()
```

Comment:
I don't think we should end the session here -- we should treat this like a syntax error. It's probably a temporary error, the user just typed something silly and they'll hit backspace soon, fixing it. We shouldn't end the session just because they made a typo.

Reply:
I think some sort of ui prompt that could let the user know about the syntax error. Similar to how when you compile a lexically/syntactically wrong program the compiler shows you where the issue is. What do you think ? I could open up a issue if you think this is something worth opening an issue for.

Reply:
We already have error highlighting for any KCL compiler errors, so this should get wired up to the CodeMirror editor's built-in diagnostics (the feature where it shows a red squiggly underline for erroneous code)